### PR TITLE
[Helix] Enable Identity.FunctionalTests

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Comment by HaoK** [comment]
Looks like copying these files over did the trick

**Comment by HaoK** [comment]
Looks like this does fail on a few helix queues due to file watcher running out of something
```
System.IO.IOException: The configured user limit (128) on the number of inotify instances has been reached, or the per-process limit on the number of open file descriptors has been reached.
   at System.IO.FileSystemWatcher.StartRaisingEvents()
   at System.IO.FileSystemWatcher.StartRaisingEventsIfNotDisposed()
   at System.IO.FileSystemWatcher.set_EnableRaisingEvents(Boolean value)
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher()
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken(String filter)
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch(String filter)
   at Microsoft.AspNetCore.Mvc.Razor.Infrastructure.DefaultFileVersionProvider.AddFileVersionToPath(PathString requestPathBase, String path) in /_/src/Mvc/Mvc.Razor/src/Infrastructure/DefaultFileVersionProvider.cs:line 48
   at Microsoft.AspNetCore.Mvc.TagHelpers.ScriptTagHelper.Process(TagHelperContext context, TagHelperOutput output) in /_/src/Mvc/Mvc.TagHelpers/src/ScriptTagHelper.cs:line 226
   at Microsoft.AspNetCore.Razor.TagHelpers.TagHelper.ProcessAsync(TagHelperContext context, TagHelperOutput output) in /_/src/Razor/Razor/src/TagHelpers/TagHelper.cs:line 55
   at Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperRunner.RunAsync(TagHelperExecutionContext executionContext) in /_/src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:line 45
   at Identity.DefaultUI.WebSite.Pages.Pages__Layout.<ExecuteAsync>b__44_1() in E:\A\_work\2\s\src\Identity\testassets\Identity.DefaultUI.WebSite\Pages\_Layout.cshtml:line 74
   at Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperExecutionContext.SetOutputContentAsync() in /_/src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperExecutionContext.cs:line 274
   at Identity.DefaultUI.WebSite.Pages.Pages__Layout.ExecuteAsync()
```

**Comment by HaoK** [comment]
File watcher issue seems similar to https://github.com/aspnet/AspNetCore/issues/7531

**Comment by HaoK** [comment]
@anurse incoming new flaky tests unfortunately, is there a way to mark the whole this whole project as flaky on some helix queues?  So it appears that there are file system issues on linux that get hit with this test.  Marking each test individually for an entire test project seems cumbersome

**Comment by HaoK** [comment]
Alternatively we could enhance SkipOnHelix to have something similar to your FlakyOn so we can stop running these tests on certain queues since the failure rate might be close to 100% on some queues

**Comment by HaoK** [comment]
https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190321.8 Failed on Fedora.27.Amd64.Open, Fedora.28.Amd64.Open, Ubuntu.1810.Amd64.Open

Rerunning to see if its the same set every time

**Comment by javiercn** [comment]
Summoning @pranavkm

I believe there's a way to turn file watching off for razor pages.

**Comment by pranavkm** [comment]
@javiercn this is from the `ScriptTagHelper`. Removing the `asp-append-version` from places like https://github.com/aspnet/AspNetCore/blob/master/src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml#L72 should disable the file watching.

**Comment by pranavkm** [comment]
@HaoK there's a couple more:

* https://github.com/aspnet/AspNetCore/blob/5a7bfbbc01920b594c468bc6e3ccf20cda97f76e/src/Identity/UI/src/Areas/Identity/Pages/V4/_Layout.cshtml#L20
* https://github.com/aspnet/AspNetCore/blob/5a7bfbbc01920b594c468bc6e3ccf20cda97f76e/src/Identity/testassets/Identity.DefaultUI.WebSite/Pages/_Layout.cshtml#L72

**Review by javiercn — APPROVED** [review]

**Comment by HaoK** [comment]
This is safe to remove from the product files right?

**Comment by javiercn** [comment]
I didn’t see the product change. It is not

**Comment by HaoK** [comment]
yeah didn't think so, so only remove it from the test one

**Comment by anurse** [comment]
> is there a way to mark the whole this whole project as flaky on some helix queues

Not right now, but I think we could do it. It looks like xUnit looks for trait attributes on the class and assembly. It would probably be as simple as changing the `AttributeUsage` of `Flaky` to allow class or assembly. I just didn't have a chance to test that out.

**Comment by HaoK** [comment]
@pranavkm doesn't look like that helped, still getting the same error with running out of inotify, (even taking out the product one) https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190321.20/workItem/Microsoft.AspNetCore.Identity.FunctionalTests-netcoreapp3.0/wilogs

**Comment by pranavkm** [comment]
The failure's from the layout page that was reverted: ` Microsoft.AspNetCore.Identity.UI.V4.Pages.Internal.Areas_Identity_Pages__Layout`. If the intent is to avoid changing product code, we could change the WebSite and replace the `WebRootFileProvider` with a `NullFileProvider` instance. That should effectively no-op the file watching.

**Comment by pranavkm** [comment]
> If the intent is to avoid changing product code, we could change the WebSite and replace the WebRootFileProvider with a NullFileProvider instance. 

Or one which no-ops the file watching. I think we still want it to return content.

**Comment by HaoK** [comment]
Well I actually didn't revert the change until this morning, so the run that failed had that change, but no-oping the file watching sounds promising

**Comment by pranavkm** [comment]
I think doing `((PhysicalFileWatcher)IHostingEnvironment.WebRootFileProvider).UseActivePolling = false` may suffice.

**Comment by HaoK** [comment]
No luck, I tried adding the line to disable `UseActivePolling` in StartupBase of the test project but same exception:

Is it possible to replace `Microsoft.AspNetCore.Mvc.Razor.Infrastructure.DefaultFileVersionProvider` to make it noop?

```
System.IO.IOException: The configured user limit (128) on the number of inotify instances has been reached, or the per-process limit on the number of open file descriptors has been reached.
   at System.IO.FileSystemWatcher.StartRaisingEvents()
   at System.IO.FileSystemWatcher.StartRaisingEventsIfNotDisposed()
   at System.IO.FileSystemWatcher.set_EnableRaisingEvents(Boolean value)
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher()
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken(String filter)
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch(String filter)
   at Microsoft.AspNetCore.Mvc.Razor.Infrastructure.DefaultFileVersionProvider.AddFileVersionToPath(PathString requestPathBase, String path) in /_/src/Mvc/Mvc.Razor/src/Infrastructure/DefaultFileVersionProvider.cs:line 48
   at Microsoft.AspNetCore.Mvc.TagHelpers.ScriptTagHelper.Process(TagHelperContext context, TagHelperOutput output) in /_/src/Mvc/Mvc.TagHelpers/src/ScriptTagHelper.cs:line 226
   at Microsoft.AspNetCore.Razor.TagHelpers.TagHelper.ProcessAsync(TagHelperContext context, TagHelperOutput output) in /_/src/Razor/Razor/src/TagHelpers/TagHelper.cs:line 55
   at Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperRunner.RunAsync(TagHelperExecutionContext executionContext) in /_/src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs:line 45
   at Microsoft.AspNetCore.Identity.UI.V4.Pages.Internal.Areas_Identity_Pages__Layout.<ExecuteAsync>b__42_11()
   at Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperExecutionContext.SetOutputContentAsync() in /_/src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperExecutionContext.cs:line 274
   at Microsoft.AspNetCore.Identity.UI.V4.Pages.Internal.Areas_Identity_Pages__Layout.<ExecuteAsync>b__42_1()
   at Microsoft.AspNetCore.Razor.Runtime.TagHelpers.TagHelperExecutionContext.SetOutputContentAsync() in /_/src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperExecutionContext.cs:line 274
   at Microsoft.AspNetCore.Identity.UI.V4.Pages.Internal.Areas_Identity_Pages__Layout.ExecuteAsync()
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context) in /_/src/Mvc/Mvc.Razor/src/RazorView.cs:line 183
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts) in /_/src/Mvc/Mvc.Razor/src/RazorView.cs:line 164
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter) in /_/src/Mvc/Mvc.Razor/src/RazorView.cs:line 298
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context) in /_/src/Mvc/Mvc.Razor/src/RazorView.cs:line 115
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode) in /_/src/Mvc/Mvc.ViewFeatures/src/ViewExecutor.cs:line 261
```

**Comment by pranavkm** [comment]
> Is it possible to replace Microsoft.AspNetCore.Mvc.Razor.Infrastructure.DefaultFileVersionProvider to make it noop?

Yup, there's an interface - https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.ViewFeatures/src/IFileVersionProvider.cs

**Comment by HaoK** [comment]
Cool replacing that appears to have done the trick, thanks @pranavkm this PR is ready for final review now, the helix tests are green

File: src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj [hunk 1]
```diff
@@ -26,4 +24,29 @@
...
+      DestinationFolder="$(PublishDir)\Identity.DefaultUI.WebSite\Pages" />
+    <Copy
+      SourceFiles="@(_wwwrootFiles)"
+      DestinationFolder="$(PublishDir)\Identity.DefaultUI.WebSite\wwwroot" />
+    <!-- Drop a dummy sln to specify content root location -->
+    <WriteLinesToFile
+        File="$(PublishDir)\contentroot.sln"
+        Lines="Ignored"
+        Overwrite="true"
+        Encoding="Unicode"/>
```
Comment (javiercn) [inline src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj]: This looks weird. I've never used it with this task, just saying.

**Review by javiercn — APPROVED** [review]
Looks good to me. :shipit: once @pranavkm is happy.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**